sdk/python/feast/feature_set.py

```diff
@@ -238,6 +256,24 @@ def created_timestamp(self, created_timestamp):
         """
         self._created_timestamp = created_timestamp
 
+    def set_label(self, key: str, value: str):
+        """
+        Sets the label value for a given key
+        """
+        if not self.labels:
+            self.labels = {key: value}
+        else:
+            self.labels[key] = value
+
+    def remove_label(self, key: str):
+        """
+        Removes a label based on key
+        """
+        if not self.labels or key not in self.labels.keys():
+            raise ValueError("Could not find label key " + key + ", no action taken")
+        elif key in self.labels.keys():
+            del self.labels[key]
```

Comment:
This is consistent with how the `drop` method behaves for features and entities on the feature set, so the `ValueError` might be the right thing for the scope of this PR. Separately though, I wonder if `KeyError` would be more appropriate for both, if not an application-specific error.

I might be inclined to initialize labels as an empty dictionary instead of the type being `Optional` though—unless there's a clear application-specific semantic reason for distinguishing no value from empty, I think empty is preferable design for collections. Give a stronger contract and spare both us and callers from null checks.

In that case, the method implementation could be reduced to `del self.labels[key]` with `KeyError` propagating, unless we want to give it a more descriptive error message.

Reply:
Indeed it makes more sense. Changed it to just use `del`. Shall I include a change for the `drop` method to raise a `KeyErrors` in this PR as well?

Reply:
Yea, makes sense.

Reply:
Do we care much that labels are an `OrderedDict`? It's not expressed by the type `MutableMapping` either. After updates we still have this signature:

```python
labels: Optional[MutableMapping[str, str]] = None
```

which to me would be nice if were simpler:

```python
labels: MutableMapping[str, str] = {}
```

Reply:
> Do we care much that labels are an `OrderedDict`? It's not expressed by the type `MutableMapping` either (this is okay if ordering is used as an implementation detail but isn't part of contract, not sure that distinction exists here though).

I don't think we care much. I just aligned this with `self._fields`. Is there a particular reason why we would want it in this case but not for labels? I can change it to initialize with a simple empty dict if this is preferred.

> After updates we still have this signature:
> 
> ```python
> labels: Optional[MutableMapping[str, str]] = None
> ```
> 
> which to me would be nice if were simpler:
> 
> ```python
> labels: MutableMapping[str, str] = {}
> ```

This would set the default value to a mutable object, which Pycharm elegantly suggests not to do:
_"Default argument values are evaluated only once at function definition time, which means that modifying the default value of the argument will affect all subsequent calls of the function."_

Reply:
Ahh yes I forgot about this quirk in Python, I'm sorry. Now I'm reminded of the idiom for argument types like `features: List[Feature] = None` and `if features is None` to initialize. So that pattern should be followed, sans `Optional` type.

Regarding `OrderedDict` I'll leave it to @woop or others, I'm not sure if there was reason behind for fields and if it should apply to labels or not.

Reply:
We do not have a special ordered contract with end users. I think when I used OrderedDict in the Python SDK I was trying to have my cake and eat it as well. 

For Fields/Features/Entities we should be using Lists (as users see them) and dicts as its implemented inside the FeatureSet class. We do not have an ordered contract with users. So technically speaking they should be maps/dicts across Feast, but from a user perspective I feel like its cleaner to expose lists, especially when persisting as YAML/JSON.

For labels I am happy to commit to simply a `dict()` instead of an OrderedDict.

Other than that the PR looks good @Joostrothweiler